#159965299 Generate JWT on user registration and login

meshack-mbuvi opened the pull request:
#### What does this PR do?
- Have JWT token on user registration and login
#### Description of Task to be completed?
- Add test for JWT token generation on user registration and login
#### How should this be manually tested?
- Clone [this repo]()
- Set up a [virtual](https://virtualenv.pypa.io/en/stable/installation/) environment for the project
- Activate it and run `pip install -r requirements` to install project dependencies
- Set up SECRET_KEY and database configuration in your environment
- Run the application using `python manage.py runserver`
- Using [Postman](https://www.getpostman.com/) navigate to 

> {your-server-root}:8000/api/users/signup/

with

> {
    "user": {"email":"your-email-here","username":"your-username","password":"your-password"}
}

#### Any background context you want to provide?
JWT authorization is efficient for user authentication. For more on JWT, click [here](https://jwt.io/introduction/)
#### What are the relevant pivotal tracker stories?
[JWT on user registration](https://www.pivotaltracker.com/story/show/159965299)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/9263906/44861118-8e6b7b00-ac80-11e8-924c-20229a8da532.png)

Commit:
[Feature #159965298] Add token generation

Commit:
[Feature #159965298] Update requirements.txt

meshack-mbuvi committed:
feat(authorization): generate jwt on user registration and login

- fix failing test and modify them to test for token generation
- generate jwt token on user registration and login

[Finishes #159965299]

pabishai commented:
Consider commenting the code you added to give more insight to the PR reviewers

pngg commented on authors/apps/authentication/token.py:
```diff
@@ -0,0 +1,16 @@
+    time = datetime.datetime.utcnow() + datetime.timedelta(seconds=86400)
+    token = jwt.encode({
+        "username": username,
+        "email": email,
```
I notice you are using both email and username as payload. I would recommend we use just the username for security and privacy concerns.

Bileonaire replied:
Thank you for the insight. Let us rectify that and use the username since it is also unique for all users. We are working on that.

Commit:
[Feature #159965298] Fix travis build

Commit:
Update tests

Commit:
Merge branch 'ft-generate-jwt-on-registration-159965299' of https://github.com/andela/ah-magnificent6 into ft-generate-jwt-on-registration-159965299
Pull changes from upstream

Commit:
[Feature #159965298] Update token payload to username only

meshack-mbuvi committed:
[Feature #159965299] Set DATABASES_URL for tests

meshack-mbuvi committed:
[Feature #159965299] Correct database password

meshack-mbuvi committed:
[Feature #159965299] Update database settings

mikenthiwa commented:
From our discussion,
Do we really need a token.py file for one function that generates token?

meshack-mbuvi commented:
@mikenthiwa From what we have discussed so far, it may not be the best approach but since there is a need to verify the token provided when users access the other endpoints, the aforementioned functionality will be done in this file unless otherwise advised by our TTL.

parseen254 commented on .travis.yml:
```diff
@@ -11,6 +11,8 @@ before_script:
   - psql -c "CREATE ROLE authorsHaven WITH SUPERUSER LOGIN PASSWORD 'authorshaven';" -U postgres
   - psql -c "CREATE DATABASE testAuthorsHaven;" -U postgres
   - export DJANGO_SETTINGS_MODULE=authors.settings
+  - export SECRET_KEY="secretkey"
```
@meshack-mbuvi push these to the environment.

meshack-mbuvi replied:
I have done it now. Please confirm that it has been done as per your expectation. Thank you.

meshack-mbuvi committed:
[Feature #159965299] Set environmental variables for testing

meshack-mbuvi committed:
[Feature #159965299] Fix error on environmental variabls

meshack-mbuvi committed:
[Feature #159965299] Fix error on environmental variables

meshack-mbuvi committed:
[Feature #159965299] Set global  environmental variables

meshack-mbuvi committed:
@meshack-mbuvi
feat(authorization): remove env varibles from travis configuration file

[Delivers #159965299]
